lib/validators.js

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+ */
+import {isDidUrl} from './did-io.js';
+import {DidResolverError} from './DidResolverError.js';
+
+/**
+ * Determines if a did contains url characters and
+ * then validates either the didUrl or did.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.did - A did.
+ *
+ * @throws {DidResolverError} Throws if did is invalid.
+ *
+ * @returns {undefined} Just returns on success.
+ */
+export function validateDid({did}) {
+  if(!did) {
+    throw new TypeError('A string "did" or "url" parameter is required.');
+  }
+  if(isDidUrl({did})) {
+    if(!isValidDidUrl({didUrl: did})) {
+      throw new DidResolverError({
+        message: `Invalid did url ${did}`,
+        code: 'invalidDidUrl'
+      });
+    }
+    return;
+  }
+  if(!isValidDid({did})) {
+    throw new DidResolverError({
+      message: `Invalid did ${did}`,
+      code: 'invalidDid'
+    });
+  }
+}
+
+/**
+ * This function comes from the did-test-suite.
```

Comment:
Would be nice to have a short one-liner actual description and put the ref to test suite below it.

Reply:
The ref to the test suite was already there in an `@see` which you probably saw, regardless I added an actual description here: https://github.com/digitalbazaar/did-io/pull/63/commits/f1f6f696881158b19ee0378b60177a05e4855c57